src/main/java/com/javadeobfuscator/deobfuscator/transformers/allatori/StringEncryptionTransformer.java

```diff
@@ -85,114 +110,174 @@ public boolean canCheckEquality(JavaValue first, JavaValue second, Context conte
+        isAllatori = isAllatori && TransformerHelper.countOccurencesOf(decryptorNode, NEWARRAY) > 0;
+        return isAllatori;
     }
 
-    private void patchMethod(Map<String, AtomicInteger> invokeCount, MethodNode method) {
-        if (invokeCount.containsKey("getStackTrace") && invokeCount.containsKey("getClassName")) {
-            for (AbstractInsnNode insn = method.instructions.getFirst(); insn != null; insn = insn.getNext()) {
-                if (insn.getOpcode() == Opcodes.NEW && (((TypeInsnNode) insn).desc.endsWith("Exception") || ((TypeInsnNode) insn).desc.endsWith("Error"))) {
-                    ((TypeInsnNode) insn).desc = "java/lang/RuntimeException";
-                } else if (insn instanceof MethodInsnNode && (((MethodInsnNode) insn).owner.endsWith("Exception") || ((MethodInsnNode) insn).owner.endsWith("Error"))) {
-                    ((MethodInsnNode) insn).owner = "java/lang/RuntimeException";
+    private static boolean isAllatoriMethod2(MethodNode decryptorNode) {
+        Map<Integer, AtomicInteger> map = TransformerHelper.calcOpcodeOccurenceMap(decryptorNode);
+        AtomicInteger zero = new AtomicInteger();
+        boolean isAllatori = true;
+        isAllatori = isAllatori && map.getOrDefault(CASTORE, zero).get() > 0;
+        isAllatori = isAllatori && map.getOrDefault(NEW, zero).get() > 0;
+        isAllatori = isAllatori && map.getOrDefault(IINC, zero).get() > 0;
+        isAllatori = isAllatori && map.getOrDefault(IXOR, zero).get() > 0;
+        isAllatori = isAllatori && map.getOrDefault(I2C, zero).get() > 0;
+        isAllatori = isAllatori && map.getOrDefault(ARETURN, zero).get() > 0;
+        isAllatori = isAllatori && map.getOrDefault(NEWARRAY, zero).get() == 1;
+        isAllatori = isAllatori && TransformerHelper.containsInvokeVirtual(decryptorNode, "java/lang/String", "charAt", "(I)C");
+        isAllatori = isAllatori && TransformerHelper.containsInvokeVirtual(decryptorNode, "java/lang/String", "length", "()I");
+        isAllatori = isAllatori && TransformerHelper.containsInvokeSpecial(decryptorNode, "java/lang/String", "<init>", null);
+        return isAllatori;
+    }
+
+    private void patchMethod(MethodNode method) {
+        boolean getStackTrace = false;
+        boolean getClassName = false;
+        for (AbstractInsnNode i = method.instructions.getFirst(); i != null; i = i.getNext()) {
+            if (!(i instanceof MethodInsnNode)) {
+                continue;
+            }
+            String name = ((MethodInsnNode) i).name;
+            if (!getStackTrace && name.equals("getStackTrace")) {
+                getStackTrace = true;
+                if (getClassName) {
+                    break;
+                }
+            } else if (!getClassName && name.equals("getClassName")) {
+                getClassName = true;
+                if (getStackTrace) {
+                    break;
+                }
+            }
+        }
+        if (!getClassName || !getStackTrace) {
```

Comment:
Maybe `if (!(getClassName && getStackTrace))`?

Reply:
I think neither representation is superior to the other, so I don't want to change it